web/src/pages/Route/components/Step1/MetaView.tsx

```diff
@@ -258,67 +269,195 @@ const MetaView: React.FC<RouteModule.Step1PassProps> = ({
     );
   };
 
-  const CustomRedirect: React.FC = () => (
-    <Form.Item
-      noStyle
-      shouldUpdate={(prev, next) => {
-        if (prev.redirectOption !== next.redirectOption) {
-          onChange({ action: 'redirectOptionChange', data: next.redirectOption });
-        }
-        return prev.redirectOption !== next.redirectOption;
-      }}
-    >
-      {() => {
-        if (form.getFieldValue('redirectOption') === 'customRedirect') {
-          return (
+  const CustomRedirectType = () => {
+    switch (form.getFieldValue('customRedirectType')) {
+      case CUSTOM_REDIRECT_TYPE.STATIC:
+        return (
+          <Form.Item
+            label={formatMessage({ id: 'page.route.form.itemLabel.redirectCustom' })}
+            required
+            style={{ marginBottom: 0 }}
+          >
+            <Row gutter={10}>
+              <Col span={5}>
+                <Form.Item
+                  name="redirectURI"
+                  rules={[
+                    {
+                      required: true,
+                      message: `${formatMessage({
+                        id: 'component.global.pleaseEnter',
+                      })}${formatMessage({
+                        id: 'page.route.form.itemLabel.redirectURI',
+                      })}`,
+                    },
+                  ]}
+                >
+                  <Input
+                    placeholder={formatMessage({
+                      id: 'page.route.input.placeholder.redirectCustom',
+                    })}
+                    disabled={disabled}
+                  />
+                </Form.Item>
+              </Col>
+              <Col span={5}>
+                <Form.Item name="ret_code" rules={[{ required: true }]}>
+                  <Select disabled={disabled} data-cy="redirect_code">
+                    <Select.Option value={301}>
+                      {formatMessage({ id: 'page.route.select.option.redirect301' })}
+                    </Select.Option>
+                    <Select.Option value={302}>
+                      {formatMessage({ id: 'page.route.select.option.redirect302' })}
+                    </Select.Option>
+                  </Select>
+                </Form.Item>
+              </Col>
+            </Row>
+          </Form.Item>
+        );
+      case CUSTOM_REDIRECT_TYPE.REGEXP:
+        return (
+          <React.Fragment>
+            <Form.List name="regex_uri" initialValue={['', '']}>
+              {(fields) =>
+                fields.map((field, index) => {
+                  switch (index) {
+                    case 0:
```

Comment:
```suggestion
                    case 0:
```
`0` and `1` Not a good choice, it makes the code somewhat unreadable